e2e/util/kustomize.go

```diff
@@ -0,0 +1,225 @@
+	if err != nil {
+		_, err = resourceInterface.Create(ctx, obj, metav1.CreateOptions{})
+		return err
+	}
+
+	obj.SetResourceVersion(existing.GetResourceVersion())
+	_, err = resourceInterface.Update(ctx, obj, metav1.UpdateOptions{})
+	return err
+}
+
+// DeleteKustomization builds and deletes resources from a kustomization using Kubernetes API
+func DeleteKustomization(kustomizationPath, namespace, kubeconfig string, wait bool) error {
+	ctx := context.Background()
+
+	yamlData, err := buildKustomization(kustomizationPath)
+	if err != nil {
+		return fmt.Errorf("failed to build kustomization: %w", err)
+	}
+
+	client, mapper, err := createK8sClient(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("failed to create Kubernetes client: %w", err)
+	}
+
+	return deleteYAML(ctx, client, mapper, yamlData, namespace)
+}
+
+// deleteYAML parses YAML and deletes resources using Kubernetes API
+func deleteYAML(ctx context.Context, client dynamic.Interface, mapper meta.RESTMapper, yamlData []byte, namespace string) error {
+	resources, err := parseYAMLResources(yamlData)
+	if err != nil {
+		return fmt.Errorf("failed to parse YAML resources: %w", err)
+	}
+
+	for _, resource := range resources {
+		if err := deleteResource(ctx, client, mapper, resource, namespace); err != nil {
+			return fmt.Errorf("failed to delete resource %s/%s: %w", resource.GetKind(), resource.GetName(), err)
+		}
+	}
+
+	fmt.Println("Deleted resources successfully")
+
+	return nil
+}
+
+func deleteResource(ctx context.Context, client dynamic.Interface, mapper meta.RESTMapper, obj *unstructured.Unstructured, namespace string) error {
+	if obj.GetNamespace() == "" && namespace != "" {
+		obj.SetNamespace(namespace)
+	}
+
+	gvk := obj.GroupVersionKind()
+	mapping, err := mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+	if err != nil {
+		return fmt.Errorf("failed to get REST mapping for %s: %w", gvk, err)
+	}
+
+	var resourceInterface dynamic.ResourceInterface
+	if mapping.Scope.Name() == meta.RESTScopeNameNamespace {
+		resourceInterface = client.Resource(mapping.Resource).Namespace(obj.GetNamespace())
+	} else {
+		resourceInterface = client.Resource(mapping.Resource)
+	}
+
+	return resourceInterface.Delete(ctx, obj.GetName(), metav1.DeleteOptions{})
+}
+
+// parseYAMLResources parses multi-document YAML into unstructured objects
+func parseYAMLResources(yamlData []byte) ([]*unstructured.Unstructured, error) {
+	var resources []*unstructured.Unstructured
+
+	decoder := yaml.NewYAMLToJSONDecoder(strings.NewReader(string(yamlData)))
+	for {
+		var rawObj runtime.RawExtension
+		if err := decoder.Decode(&rawObj); err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, fmt.Errorf("failed to decode YAML: %w", err)
+		}
+
+		if len(rawObj.Raw) == 0 {
+			continue
+		}
+
+		obj := &unstructured.Unstructured{}
+		if err := obj.UnmarshalJSON(rawObj.Raw); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal JSON: %w", err)
+		}
```

Comment:
Decoding yaml to json and json to unstructured.Unstructured{} is not great. Ideally we can get json or []unstructured.Unstructured{} from kustomize and avoid this code.